src/pages/Account/Profile.tsx

```diff
@@ -2,42 +2,36 @@ import React from "react";
 import { useForm } from "react-hook-form";
 import { useHistory } from "react-router-dom";
 import "firebase/storage";
+import { useAsyncFn, useSearchParam } from "react-use";
 
 import { IS_BURN } from "secrets";
 
 import { DEFAULT_VENUE, DISPLAY_NAME_MAX_CHAR_COUNT } from "settings";
 
-import { RouterLocation } from "types/RouterLocation";
-
-import getQueryParameters from "utils/getQueryParameters";
-
 import { useVenueId } from "hooks/useVenueId";
 import { useUser } from "hooks/useUser";
 
-import { updateUserProfile } from "./helpers";
-
+import { Loading } from "components/molecules/Loading";
 import { ProfilePictureInput } from "components/molecules/ProfilePictureInput";
 
+import { updateUserProfile } from "./helpers";
+
+// @debt refactor the Profile related styles from Account.scss into Profile.scss
 import "./Account.scss";
 
 export interface ProfileFormData {
   partyName: string;
   pictureUrl: string;
 }
 
-interface PropsType {
-  location?: RouterLocation;
-}
-
-const Profile: React.FunctionComponent<PropsType> = ({ location }) => {
+export const Profile: React.FC = () => {
   const history = useHistory();
-  const { user } = useUser();
-  const venueName = useVenueId();
-  const venueId =
-    venueName ??
-    getQueryParameters(window.location.search)?.venueId?.toString() ??
-    DEFAULT_VENUE;
-  const { returnUrl } = getQueryParameters(window.location.search);
+  const { user, userWithId } = useUser();
+
+  const venueId = useVenueId() ?? DEFAULT_VENUE;
+
+  const returnUrl: string | undefined =
+    useSearchParam("returnUrl") ?? undefined;
```

Comment:
can it return null?

Reply:
It does by default, and I _guess_ it can since `??` handles `null` as well. But given we tend to use `undefined` I wanted to align it to our standards as soon as possible.